modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/70_execute_painless_scripts.yml

```diff
@@ -68,3 +74,108 @@ setup:
               rank: 4
             index: "my-index"
   - match: { result: 0.8 }
+
+---
+"Execute with boolean field context":
+  - do:
+      scripts_painless_execute:
+        body:
+          script:
+            source: "emit(doc['rank'].value < params.max_rank)"
+            params:
+              max_rank: 5.0
+          context: "boolean_script_field_script_field"
+          context_setup:
+            document:
+              rank: 4
+            index: "my-index"
+  - match: { result: { trues: 1, falses: 0 } }
+
+---
+"Execute with date field context":
+  - do:
+      scripts_painless_execute:
+        body:
+          script:
+            source: "emit(doc['date'].value.toInstant().toEpochMilli())"
+          context: "double_script_field_script_field"
+          context_setup:
+            document:
+              date: "2015-01-01T12:10:30Z"
+            index: "my-index"
+  - match: { result: [ 1.42011423E12 ] }
+
+---
+"Execute with double field context":
+  - do:
+      scripts_painless_execute:
+        body:
+          script:
+            source: "emit(doc['rank'].value * params.max_rank)"
+            params:
+              max_rank: 5.0
+          context: "double_script_field_script_field"
+          context_setup:
+            document:
+              rank: 4
+            index: "my-index"
+  - match: { result: [ 20.0 ] }
+
+---
+"Execute with geo point field context":
+  - do:
+      scripts_painless_execute:
+        body:
+          script:
+            source: "emit(doc['point'].value.lat + 1.0, doc['point'].value.lon - 1.0)"
+          context: "geo_point_script_field_script_field"
+          context_setup:
+            document:
+              point: "30.0,40.0"
+            index: "my-index"
+  - match: { result: [ 3176939252927413179 ] }
+
+---
+"Execute with ip field context":
+  - do:
+      scripts_painless_execute:
+        body:
+          script:
+            source: "emit(doc['ip'].value);"
+          context: "ip_script_field_script_field"
+          context_setup:
+            document:
+              ip: "192.168.1.254"
+            index: "my-index"
+
+---
+"Execute with long field context":
+  - do:
+      scripts_painless_execute:
+        body:
+          script:
+            source: "emit(doc['rank'].value * (long)params.max_rank)"
+            params:
+              max_rank: 5.0
+          context: "long_script_field_script_field"
+          context_setup:
+            document:
+              rank: 4
+            index: "my-index"
+  - match: { result: [ 20 ] }
```

Comment:
I think it'd be nice to make sure that multi-valued results work with this API. Folks'll use them less than single-valued results so we might not catch future bugs that break them until they are out in the wild.

Reply:
@nik9000 @javanna Based on the boolean feedback, when the results come back should I be ordering them to match what doc values would do then as well for others?

Reply:
Yeah - I'd sort like doc_values does. It might not be right, but at least it lines up with what we do in other places.

Reply:
I would add a method that does it to BOoleanFieldScript, I am doing the same as part of another task that is not a PR yet. you got to it first :)